pallets/validators-manager/src/tests/tests.rs

```diff
@@ -274,9 +307,14 @@ mod remove_validator_public {
                 ]
             );
 
-            // Advance 2 sessions
-            advance_session();
-            advance_session();
+            // Advance sessions to trigger removal via session handler
+            // The session handler needs to see the validator removed from active set,
+            // then clean_up_collator_data calls execute_leave_candidates,
+            // which schedules the exit (takes multiple sessions)
+            advance_session(); // Session 1: on_new_session triggers clean_up_collator_data
+            advance_session(); // Session 2: execute_leave_candidates completes
+            advance_session(); // Session 3: ParachainStaking removes from candidate pool
+            advance_session(); // Session 4: Session updates validator set
```

Comment:
I think there is a helper function somewhere that you can feed the number of sessions you want to advance